src/PlotlyPanel.tsx

```diff
@@ -169,54 +172,80 @@ export const PlotlyPanel: React.FC<Props> = (props) => {
 const setDataFrameId = (frame: DataFrame) => {
   if (frame.meta?.custom?.id) {
     return;
-  } else if (frame.meta?.custom) {
-    frame.meta.custom['id'] = getGuid();
-  } else if (frame.meta) {
-    frame.meta.custom = { id: getGuid() };
-  } else {
-    frame.meta = { custom: { id: getGuid() } };
   }
+  _.set(frame, 'meta.custom.id', _.uniqueId());
 };
 
-// Pull out user-selected fields from a DataFrame.
-// If the options aren't set or the fields don't exist, try to pick the first
-// available fields, prioritizing time fields on the X axis.
-const getFields = (frame: DataFrame, props: Props) => {
-  let xField = frame.fields.find((field) => field.name === props.options.xAxis.field);
-  if (!xField) {
-    xField = frame.fields.find((field) => field.type === FieldType.time);
-    if (!xField) {
-      const y = props.options.yAxis.fields || [];
-      xField = frame.fields.find((field) => !y.includes(field.name));
+const getXFields = (frames: DataFrame[], selectedField?: string) => {
```

Comment:
This function does some validation and implements the logic for selecting which fields to put on the X Axis, looking at all fields across N data frames. We either try to find the user-selected fields, or pick them automatically with this heuristic (one field per input data frame):
Time fields of the same name 
String fields of the same name
Time fields of any name 
Numeric fields of the same name